charts/nginx-gateway-fabric/templates/clusterrole.yaml

```
@@ -8,13 +8,15 @@ rules:
 - apiGroups:
   - ""
   - apps
+  - autoscaling
```

Let's only add this if autoscaling is enabled
```suggestion
  {{- if or .Values.nginx.autoscaling.enabled .Values.nginxGateway.autoscaling.enabled }}
  - autoscaling
  {{- end }}
```